qiskit/circuit/library/data_preparation/state_preparation.py

```diff
@@ -0,0 +1,502 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Prepare a quantum state from the state where all qubits are 0."""
+
+from typing import Union, Optional
+
+import math
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.library.standard_gates.x import CXGate, XGate
+from qiskit.circuit.library.standard_gates.h import HGate
+from qiskit.circuit.library.standard_gates.s import SGate, SdgGate
+from qiskit.circuit.library.standard_gates.ry import RYGate
+from qiskit.circuit.library.standard_gates.rz import RZGate
+from qiskit.circuit.exceptions import CircuitError
+from qiskit.quantum_info import Statevector
+
+_EPS = 1e-10  # global variable used to chop very small numbers to zero
+
+
+class StatePreparation(Gate):
+    """Complex amplitude state preparation.
+
+    Class that implements the (complex amplitude) state preparation of some
+    flexible collection of qubit registers.
+    """
+
+    def __init__(
+        self,
+        params: Union[str, list, Statevector],
+        num_qubits: int = None,
+        inverse: bool = False,
+        label: Optional[str] = "State Preparation",
+    ):
+        r"""
+        Args:
+            params (str, list, int or Statevector):
```

Comment:
Could you remove the type hints from the docstring, now that they are in the signature? 🙂